### ISSUE TITLE
Simultaneous MIDI BLE notes not played.

### ISSUE THREAD

**Issue by JerryH31** [adafruit/Adafruit_CircuitPython_BLE_MIDI#2]
When a MIDI file contains simultaneous notes (like a chord), only one of the notes is played. I suspect this has something to do with MIDI 'running status' not being implemented in the BLE MIDI Library.

Note that the circuitpython MIDI library (MIDI over USB) handles simultaneous notes correctly.

I've attached a test file that plays 4 G chords:
[G chord test (short).zip](https://github.com/adafruit/Adafruit_CircuitPython_BLE_MIDI/files/4465182/G.chord.test.short.zip)

**Comment by tannewt** [comment]
I agree this is a running status issue. It's non-trivial to support because the BLE MIDI spec doesn't allow running status to run across packets.

![Screen Shot 2020-04-13 at 4 17 40 PM](https://user-images.githubusercontent.com/52649/79169830-53a94e80-7da2-11ea-9080-92a5ab9b571d.png)

It should be doable but isn't a high priority for me.

**Comment by JerryH31** [comment]
Oh, I didn't realize it was difficult to implement this.  Maybe referencing the Arduino Bluefruit MIDI implementation might provide some insight:  
https://learn.adafruit.com/web-ble-midi/ble-midi
I know firsthand that the Arduino Bluefruit MIDI implementation works well--no MIDI notes are dropped.  I wish I could help, but this is way beyond my skill level.

**Comment by tannewt** [comment]
The Arduino implementation doesn't merge midi messages into a single packet so I'm not sure why it works there. Maybe because it always inserts timestamps? Here is what I was looking at for the Arduino implementation: https://github.com/adafruit/Adafruit_nRF52_Arduino/blob/master/libraries/Bluefruit52Lib/src/services/BLEMidi.cpp

I'm going to reopen in case someone else wants to fix it.

**Comment by JerryH31** [comment]
Just wondering if anyone has an interest in fixing this issue.  My project is stalled without the ability of this code to handle MIDI 'Running Status' messages. I wish I could work on the code myself, but it is beyond my abilities.  ...I'm certainly willing to test and provide feedback!

**Comment by AllTheProjects** [comment]
This is a major roadblock for me as well. My project is dead in the water.

**Comment by ladyada** [comment]
best bet is to send individual note on/off's for now

**Comment by AllTheProjects** [comment]
> best bet is to send individual note on/off's for now

Wish I could. It was supposed to read MIDI notes out of the Synesthesia Game piano app. I am unable to change the way the app sends out MIDI.

**Comment by JerryH31** [comment]
I'm still hoping someone might fix this issue. I don't have control over the source MIDI material. My MIDI source uses running status messages.

**Comment by keithngroover** [comment]
I found a way to send multiple messages in one packet, but it feels like a hack to me.

First off, [this code on the adafruit site](https://learn.adafruit.com/adafruit-feather-m0-bluefruit-le/ble-services#at-plus-blemiditx-2435789) straight up doesn't work:

`# Send 2 events
AT+BLEMIDITX=90-3C-7F-A0-3C-7F
OK`

The problem is that synths expect a timestamp in between each event. So what I did was fake a timestamp by putting an EE in there, like this:

`AT+BLEMIDITX=90-3C-7F-EE-A0-3C-7F`

I have a MIDI controller that needs to stop and start multiple notes simultaneously, so I wrote up this monstrosity:

`//turning off old notes
ble.print((128 + MIDIchannel), HEX);
ble.print("-");
lastNote = constrain(lastNote, 0, 127);
if (lastNote <=15)
{
  ble.print("0");
  ble.print(lastNote, HEX);
}
else
  {ble.print(lastNote, HEX);}
if (pcMode == true)
{
ble.print("-00-EE-"); //fake timestamp

//turning off old power chord note
ble.print((128 + MIDIchannel), HEX);
ble.print("-");
//lastNote = constrain(lastNote, 0, 127);
if ((lastNote + 7) <=15)
{
  ble.print("0");
  ble.print((lastNote + 7), HEX);
}
else
  {ble.print((lastNote + 7), HEX);}
}
ble.println("-00");
 //end of first packet`

Whenever print formats HEX, it doesn't always print two digits, which is why there's that if (lastNote <=15) stuff in there.

It's ugly, but it works. If there's a better and fast way to do this, I'd love to see it.

**Comment by appliko** [comment]
I may have similar issue.
I send 4 CC's to my device as commands, and I can read only the first one by msg.receive()
From sender log, I think are sent on a single command
Is BLE midi library capable to handle multiple CC on a single command ?
